storage/remote/write_test.go

```diff
@@ -476,3 +480,54 @@ func generateOTLPWriteRequest() pmetricotlp.ExportRequest {
 
 	return pmetricotlp.NewExportRequestFromMetrics(d)
 }
+
+func TestOTLPDelta(t *testing.T) {
+	var logbuf bytes.Buffer
+	log := slog.New(slog.NewTextHandler(&logbuf, &slog.HandlerOptions{Level: slog.LevelWarn}))
+
+	appendable := &mockAppendable{}
+	handler := NewOTLPWriteHandler(log, nil, appendable, func() config.Config {
+		return config.Config{
+			OTLPConfig: config.DefaultOTLPConfig,
+		}
+	}).(*otlpWriteHandler)
+
+	md := pmetric.NewMetrics()
+	ms := md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics()
+
+	m := ms.AppendEmpty()
+	m.SetName("some.delta.total")
+
+	sum := m.SetEmptySum()
+	sum.SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+
+	ts := time.Date(2000, 1, 2, 3, 4, 0, 0, time.UTC)
+	for i := range 3 {
+		dp := sum.DataPoints().AppendEmpty()
+		dp.SetIntValue(int64(i))
+		dp.SetTimestamp(pcommon.NewTimestampFromTime(ts.Add(time.Duration(i) * time.Second)))
+	}
```

Comment:
I'm not sure about that.
The existing prometheus code does not return an error if it encounters a delta sample, but instead only logs it.
Still returns http 200.

My test relies not "no log output", which is somewhat brittle in its own, but I think acceptable here because it does very limited processing.

Doing the same on the big test could make it very flaky, which I'd like to aovid

Reply:
Instead of asserting on no warning logs, couldn't we verify that mockappendable has metrics as expected?